back/app/controllers/omniauth_callback_controller.rb

```diff
@@ -187,7 +187,7 @@ def update_user!(auth, user, authver_method)
     attrs = authver_method.updateable_user_attrs
     user_params = authver_method.profile_to_user_attrs(auth).slice(*attrs).compact
     user_params.delete(:remote_avatar_url) if user.avatar.present? # don't overwrite avatar if already present
-    confirm_user = authver_method.email_confirmed?(auth)
+    confirm_user = authver_method.email_confirmed?(auth) && user_params[:email].present?
```

Comment:
See the commit description for the details https://github.com/CitizenLabDotCo/citizenlab/pull/6152/commits/cdb2b9501923c4f26b2e005e83b9963a9fa29695

> Currently, when a user signs in with SSO, we find it not only by email, but also by the provider UID (stored in Identity or in Verification). If we find a user by the provider UID, and the provider doesn't return email, we don't want to confirm the user's email.

Reply:
Though it should not happen in practice because if `email_confirmed?` returns true, `email` should be present. But it can happen if some provider (e.g., facebook) stops sending us email (maybe it happens currently, not sure) and `email_confirmed?` always returns true for this provider.

Reply:
Are we saying that the provider could return email verified, without sending us the email? But if this happens then likely they'll find themselves locked out of doing much on our platform. But I guess this is fine so long as it doesn't actually happen in practice

Reply:
> Are we saying that the provider could return email verified, without sending us the email?

By default,  we just suppose that email is verified without really checking it https://github.com/CitizenLabDotCo/citizenlab/blob/d1eff334123e74e9c7f5f295b659c75656a9e6c7/back/lib/omniauth_methods/base.rb#L52-L54
In most cases, it should be fine though.

And in some cases (Hoplr and Google - see this PR's code) we actually check the value the provider sends us.

I've also changed the logic a bit and added more tests here https://github.com/CitizenLabDotCo/citizenlab/pull/6152/commits/d1eff334123e74e9c7f5f295b659c75656a9e6c7